app/controllers/api/v2/provisioning_templates_controller.rb

```diff
@@ -0,0 +1,133 @@
+                :desc => N_("Array of template combinations (hostgroup_id, environment_id)")
+          param :operatingsystem_ids, Array, :desc => N_("Array of operating system IDs to associate with the template")
+          param :locked, :bool, :desc => N_("Whether or not the template is locked for editing")
+          param_group :taxonomies, ::Api::V2::BaseController
+        end
+      end
+
+      api :POST, "/provisioning_templates/", N_("Create a provisioning template")
+      param_group :provisioning_template, :as => :create
+
+      def create
+        @provisioning_template = ProvisioningTemplate.new(params[:provisioning_template])
+        process_response @provisioning_template.save
+      end
+
+      api :PUT, "/provisioning_templates/:id", N_("Update a provisioning template")
+      param :id, :identifier, :required => true
+      param_group :provisioning_template
+
+      def update
+        process_response @provisioning_template.update_attributes(params[:provisioning_template])
+      end
+
+      api :GET, "/provisioning_templates/revision"
+      param :version, String, :desc => N_("template version")
+
+      def revision
+        audit = Audit.authorized(:view_audit_logs).find(params[:version])
+        render :json => audit.revision.template
+      end
+
+      api :DELETE, "/provisioning_templates/:id", N_("Delete a provisioning template")
+      param :id, :identifier, :required => true
+
+      def destroy
+        process_response @provisioning_template.destroy
+      end
+
+      api :GET, "/provisioning_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
+
+      def build_pxe_default
+        status, msg = ProvisioningTemplate.authorized(:deploy_provisioning_templates).build_pxe_default(self)
+        render :json => msg, :status => status
+      end
+
+      def_param_group :provisioning_template_clone do
+        param :provisioning_template, Hash, :required => true, :action_aware => true do
+          param :name, String, :required => true, :desc => N_("template name")
+        end
+      end
+
+      api :POST, "/provisioning_templates/:id/clone", N_("Clone a provision template")
+      param :id, :identifier, :required => true
+      param_group :provisioning_template_clone, :as => :create
+
+      def clone
+        @provisioning_template = @provisioning_template.clone
+        load_vars_from_template
+        @provisioning_template.name = params[:provisioning_template][:name]
+        process_response @provisioning_template.save
+      end
+
+      def resource_name
+        'provisioning_template'
+      end
+
+      private
+
+      def resource_class
+        ProvisioningTemplate
+      end
+
+      def process_operatingsystems
+        return unless (ct = params[:provisioning_template]) and (operatingsystems = ct.delete(:operatingsystems))
+        ct[:operatingsystem_ids] = operatingsystems.collect {|os| os[:id]}
```

Comment:
what does it even stand for: it should not be the config template anymore, so any other abbreviation I know is 'Czech Television' :)